File: tools/inspect_all_data.mjs
import { readFileSync } from 'node:fs';

const dataSrc = readFileSync('assets/js/data.js', 'utf8');
const ctx = {};
new Function(`${dataSrc}; this.VOLUMES=VOLUMES; this.COLLECTION=COLLECTION;`).call(ctx);
const { COLLECTION } = ctx;

console.log(`Total items in COLLECTION: ${COLLECTION.length}`);
const kp = COLLECTION.filter(d => d.v === 'kala-parampara');
const kr = COLLECTION.filter(d => d.v === 'kala-rasa');

console.log(`\n=== KALA PARAMPARA (${kp.length} items) ===`);
kp.forEach((d, i) => {
  console.log(`${i+1}. [${d.id}] ${d.no} | "${d.n}" | sub: "${d.sub || ''}" | img: ${d.img}`);
});

console.log(`\n=== KALA RASA (${kr.length} items) ===`);
kr.forEach((d, i) => {
  console.log(`${i+1}. [${d.id}] ${d.no} | "${d.n}" | sub: "${d.sub || ''}" | img: ${d.img}`);
});

File: tools/inspect_all_data.py
import json
import re

with open('assets/js/data.js', 'r', encoding='utf-8') as f:
    js_text = f.read()

# Let's extract COLLECTION array
match = re.search(r'const COLLECTION\s*=\s*(\[.*?\]);', js_text, re.DOTALL)
if match:
    # Convert JS object literal to JSON
    json_like = match.group(1)
    # quote unquoted keys
    json_like = re.sub(r'(\s*)(id|v|n|no|sub|b|img|sp|cat|tag):', r'\1"\2":', json_like)
    # remove trailing commas
    json_like = re.sub(r',\s*([\]}])', r'\1', json_like)
    data = json.loads(json_like)
    print(f"Loaded {len(data)} items from data.js")
    
    kp_items = [d for d in data if d.get('v') == 'kala-parampara']
    kr_items = [d for d in data if d.get('v') == 'kala-rasa']
    print(f"Kala Parampara items: {len(kp_items)}")
    print(f"Kala Rasa items: {len(kr_items)}")
    
    print("\n--- First 15 Kala Parampara Items ---")
    for d in kp_items[:15]:
        print(f"{d['id']} | {d['no']} | {d['n']} | {d.get('sub', '')} | img: {d['img']}")
        
    print("\n--- First 15 Kala Rasa Items ---")
    for d in kr_items[:15]:
        print(f"{d['id']} | {d['no']} | {d['n']} | {d.get('sub', '')} | img: {d['img']}")
